File: ClimbEdge.Application/DTOs/BoardDTO.cs
using ClimbEdge.Domain.Enums.Boards;

namespace ClimbEdge.Application.DTOs
{
    public record class CreateBoardDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public BoardVisibility Visibility { get; set; } = BoardVisibility.Private;
        public long BoardConfigId { get; set; }
        public long? OrganizationId { get; set; }
    }

    public record class UpdateBoardDTO
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public BoardVisibility? Visibility { get; set; }
        public long? OrganizationId { get; set; }
    }

    public record class GetBoardDTO : BaseDTO
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public BoardVisibility Visibility { get; set; }
        public long BoardConfigId { get; set; }
        public long? OrganizationId { get; set; }
    }

    public record class AddBoardMemberDTO
    {
        public long BoardId { get; set; }
        public long UserId { get; set; }
        public BoardMemberRole Role { get; set; } = BoardMemberRole.Member;
    }

    public record class GetBoardMemberDTO : BaseDTO
    {
        public long BoardId { get; set; }
        public long UserId { get; set; }
        public BoardMemberRole Role { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/BoardDTO.ts
import { BaseDTO } from './BaseDTO';

// Enums
export enum BoardVisibility {
  Private = 0,
  Organization = 1,
  Public = 2,
}

export enum BoardMemberRole {
  Member = 0,
  Admin = 1,
  Owner = 2,
}

// Create / Update
export interface CreateBoardDTO {
  name: string;
  description?: string;
  visibility: BoardVisibility;
  boardConfigId: number;
  organizationId?: number | null;
}

export interface UpdateBoardDTO {
  name?: string;
  description?: string;
  visibility?: BoardVisibility;
  organizationId?: number | null;
}

export interface AddBoardMemberDTO {
  boardId: number;
  userId: number;
  role: BoardMemberRole;
}

// Read
export class GetBoardDTO extends BaseDTO {
  name!: string;
  description?: string;
  visibility!: BoardVisibility;
  boardConfigId!: number;
  organizationId?: number | null;
}

export class GetBoardMemberDTO extends BaseDTO {
  boardId!: number;
  userId!: number;
  role!: BoardMemberRole;
}

// Filters
export interface BoardFilters {
  organizationId?: number;
  page?: number;
  pageSize?: number;
}

export interface BoardProblemFilters {
  includeArchived?: boolean;
  page?: number;
  pageSize?: number;
}
